go/vt/vtgate/engine/rename_fields.go

```diff
@@ -73,9 +73,12 @@ func (r *RenameFields) Execute(vcursor VCursor, bindVars map[string]*querypb.Bin
 }
 
 func (r *RenameFields) renameFields(qr *sqltypes.Result) {
-	for i := 0; i < len(r.Indices) && i < len(qr.Fields); i++ {
-		colName := r.Cols[r.Indices[i]]
-		qr.Fields[i].Name = colName
+	for ind, index := range r.Indices {
+		if ind == len(qr.Fields) {
+			break
+		}
+		colName := r.Cols[ind]
+		qr.Fields[index].Name = colName
 	}
```

Comment:
this will panic for case 
when qr.Fields has 5 columns
and the r.Indices want to replace 2 and 6 index fields.

Reply:
my bad, pushed the fix